pkg/autodiscovery/listeners/cloudfoundry_test.go

```diff
@@ -0,0 +1,417 @@
+							"variables":    json.RawMessage(`{"host":"$.credentials.host","username":"$.credentials.Username","password":"$.credentials.Password","dbname":"$.credentials.database_name"}`),
+						},
+						"flask-app": map[string]json.RawMessage{
+							"check_names":  json.RawMessage(`["http_check"]`),
+							"init_configs": json.RawMessage(`[{}]`),
+							"instances":    json.RawMessage(`[{"name": "My Nginx", "url": "http://%%host%%:%%port_p8080%%", "timeout": 1}]`),
+						},
+					},
+					EnvVcapServices: map[string][]byte{"my-postgres": []byte(`{"credentials":{"host":"a.b.c","Username":"me","Password":"secret","database_name":"mydb"}}`)},
+				},
+				{
+					AppGUID:     "appguid3",
+					ProcessGUID: "processguid3",
+				},
+			},
+			expDel: map[string]Service{
+				"myprocessguid1/my-postgres": &CloudFoundryService{
+					containerIPs:   map[string]string{},
+					containerPorts: []ContainerPort{},
+					creationTime:   integration.After,
+				},
+			},
+			expNew: map[string]Service{
+				"processguid1/flask-app/0": &CloudFoundryService{
+					containerIPs: map[string]string{CfServiceContainerIP: "1.2.3.4"},
+					containerPorts: []ContainerPort{
+						{
+							Name: "p11",
+							Port: 11,
+						},
+						{
+							Name: "p22",
+							Port: 22,
+						},
+					},
+					creationTime: integration.After,
+				},
+				"processguid1/flask-app/1": &CloudFoundryService{
+					containerIPs: map[string]string{CfServiceContainerIP: "1.2.3.5"},
+					containerPorts: []ContainerPort{
+						{
+							Name: "p33",
+							Port: 33,
+						},
+						{
+							Name: "p44",
+							Port: 44,
+						},
+					},
+					creationTime: integration.After,
+				},
+				"processguid1/my-postgres": &CloudFoundryService{
+					containerIPs:   map[string]string{},
+					containerPorts: []ContainerPort{},
+					creationTime:   integration.After,
+				},
+				"processguid2/flask-app/0": &CloudFoundryService{
+					containerIPs: map[string]string{CfServiceContainerIP: "1.2.3.7"},
+					containerPorts: []ContainerPort{
+						{
+							Name: "p77",
+							Port: 77,
+						},
+						{
+							Name: "p88",
+							Port: 88,
+						},
+					},
+					creationTime: integration.After,
+				},
+				"processguid2/flask-app/1": &CloudFoundryService{
+					containerIPs: map[string]string{CfServiceContainerIP: "1.2.3.8"},
+					containerPorts: []ContainerPort{
+						{
+							Name: "p99",
+							Port: 99,
+						},
+						{
+							Name: "p111",
+							Port: 111,
+						},
+					},
+					creationTime: integration.After,
+				},
+				"processguid2/my-postgres": &CloudFoundryService{
+					containerIPs:   map[string]string{},
+					containerPorts: []ContainerPort{},
+					creationTime:   integration.After,
+				},
+			},
+		},
+	} {
+		// NOTE: we don't use t.Run here, since the executions are chained (every tc is expected to delete some services
```

Comment:
what does tc means?

Reply:
Test case - I'll update the comment to prevent confusion.